File: 2-javascript/advanced/mapDebris.js
function orbitalPeriod (arr) {
  var GM = 398600.4418;
  var radius = 6367.4447;
  var result = [];

  for (var index in arr) {
    result.push({
      name: arr[index].name,
      orbitalPeriod: getOP(arr[index].avgAlt, GM, radius)
    });
  }

  return result;
}

function getOP (alt, GM, radius) {
  var op = 2 * Math.PI * Math.sqrt((Math.pow(alt + radius, 3) / GM));
  return Math.round(op);
}

console.log(orbitalPeriod([{
  name: 'sputnik',
  avgAlt: 35873.5553
}]));

console.log(orbitalPeriod([{
  name: 'iss',
  avgAlt: 413.6
}, {
  name: 'hubble',
  avgAlt: 556.7
}, {
  name: 'moon',
  avgAlt: 378632.553
}]));

File: old-beta/2-algorithm/interGo/mapDebris.go
package main

import "fmt"
import "math"

func mapDebris() {
	var total = make([]sol, len(test07))
	for i, v := range test07 {
		var s sol
		for k, t := range v.data {
			fmt.Sprintln(i, v, k, t)
			var aux response
			aux.name = t.name
			aux.orbitalPeriod = getOrbitalPeriod(t.avgAlt)
			s = append(s, aux)
		}
		total[i] = s
	}
	for i, v := range total {
		fmt.Println(i, v)
	}
}

func getOrbitalPeriod(alt float64) int {
	var gm = 398600.4418
	var radius = 6367.4447
	h := 2 * math.Pi * math.Sqrt((math.Pow(alt+radius, 3) / gm))
	return roundFloat64(h)
}

func roundFloat64(num float64) int {
	if num < 0 {
		return int(num - 0.5)
	}
	return int(num + 0.5)
}

type response struct {
	name          string
	orbitalPeriod int
}

type sol []response

type entry struct {
	name   string
	avgAlt float64
}

var test07 = []struct {
	data []entry
}{
	{
		data: []entry{
			{name: "sputnik", avgAlt: 35873.5553},
		},
	},
	{
		data: []entry{
			{name: "iss", avgAlt: 413.6},
			{name: "hubble", avgAlt: 556.7},
			{name: "moon", avgAlt: 378632.553},
		},
	},
}
